jars/src/org/toast/activegates/GateCommandPlugin.kt

```diff
@@ -0,0 +1,93 @@
+            if (cost < 1) cost = 1f
+            return cost
+        }
+
+    val commodityCostString: String
+        get() {
+            val cargo = Global.getSector().playerFleet.cargo
+            return ("${getCommodityCost("metals").roundToInt()} (${cargo.getCommodityQuantity("metals").roundToInt()}) metals, " +
+                    "${getCommodityCost("heavy_machinery").roundToInt()} (${cargo.getCommodityQuantity("heavy_machinery").roundToInt()}) heavy machinery, " +
+                    "and some kind of basic processing core")
+        }
+
+    private val activationCost: Map<String, Float> =
+            mapOf(
+                    "metals" to Global.getSettings().getFloat("activeGates_Metals"),
+                    "heavy_machinery" to Global.getSettings().getFloat("activeGates_HeavyMachinery"),
+                    "gamma_core" to Global.getSettings().getFloat("activeGates_GammaCores"))
+
+    fun canActivate(): Boolean {
+        val cargo = Global.getSector().playerFleet.cargo
+        return activationCost.all { commodityAndCost -> cargo.getCommodityQuantity(commodityAndCost.key) >= commodityAndCost.value }
+    }
+
+    fun payActivationCost(): Boolean {
+        val cargo = Global.getSector().playerFleet.cargo
+        return if (canActivate()) {
+            activationCost.forEach { commodityAndCost ->
+                cargo.removeCommodity(commodityAndCost.key, commodityAndCost.value)
+            }
+            true
+        } else {
+            false
+        }
+    }
+
+    /**
+     * List of systems with gates (filterable), sorted by shortest distance from player first
+     */
+    fun getGateMap(filter: GateFilter, excludeCurrentGate: Boolean = true): List<GateDestination> {
+        val playerLoc = Global.getSector().playerFleet.locationInHyperspace
+
+        return Global.getSector().starSystems
+                .filter {
+                    when (filter) {
+                        GateFilter.Active -> it.getEntitiesWithTag(TAG_GATE_ACTIVATED).any()
+                        GateFilter.Inactive -> it.getEntitiesWithTag(TAG_GATE).any() && it.getEntitiesWithTag(TAG_GATE_ACTIVATED).none()
+                        GateFilter.All -> it.getEntitiesWithTag(TAG_GATE).any()
+                    }
+                }
+                .map {
+                    GateDestination(
+                            systemId = it.id,
+                            systemName = it.baseName,
+                            distanceFromPlayer = Misc.getDistanceLY(playerLoc, it.location))
+                }
+                .filter {
+                    if (excludeCurrentGate)
+                        it.distanceFromPlayer > 0
+                    else
+                        true
+                }
+                .sortedBy { it.distanceFromPlayer }
+    }
+
+    private fun getCommodityCost(commodity: String): Float = activationCost[commodity] ?: 0F
+
+    const val TAG_GATE_ACTIVATED = "gate_activated"
```

Comment:
maybe we should move away from "activate" and "activated" to "bring online" and "online", following the phrasing for comm relays, nav beacons, and sensor arrays.